packages/zoe/src/contracts/priceAggregator.js

```diff
@@ -231,19 +241,75 @@ const start = async zcf => {
       });
       ({ priceAuthority, adminFacet: priceAuthorityAdmin } = paKit);
     },
-    async initOracle(oracleInstance, query = undefined) {
+    makeOracleInvitation: async () => {
+      /** @type {OfferHandler<OracleAdmin>} */
+      const offerHandler = async (
+        seat,
+        { notifier: oracleNotifier, scaleValueOut = 1 } = {},
+      ) => {
+        const admin = await creatorFacet.initOracle();
+        seat.exit();
+        if (!oracleNotifier) {
+          // No notifier to track, just let them have the direct admin.
+          return admin;
+        }
+
+        // Adapt the notifier to push results.
+        const recurse = ({ value, updateCount }) => {
+          if (!oracleNotifier || !updateCount) {
+            // Interrupt the cycle because we either are deleted or the notifier
+            // finished.
+            return;
+          }
+          // Queue the next update.
+          E(oracleNotifier).getUpdateSince(updateCount).then(recurse);
+
+          // See if we have associated parameters or just a raw value.
+          const data = value.data || value;
+
+          // Push the current scaled result.
+          const scaledData = Math.floor(parseInt(data, 10) * scaleValueOut);
+          const newData = BigInt(scaledData);
+
+          if (value.data) {
+            // We have some associated parameters to push.
+            const newValue = { ...value, data: newData };
+            admin.pushResult(newValue).catch(console.error);
+          } else {
+            admin.pushResult(newData).catch(console.error);
+          }
+        };
+
+        // Start the notifier.
+        E(oracleNotifier).getUpdateSince().then(recurse);
+
+        return Far('oracleAdmin', {
+          ...admin,
+          delete: async () => {
+            // Stop tracking the notifier on delete.
+            oracleNotifier = undefined;
+            return admin.delete();
+          },
+        });
+      };
+
+      return zcf.makeInvitation(offerHandler, 'oracle invitation');
+    },
+    async initOracle(oracleInstance = undefined, query = undefined) {
       assert(quoteKit, X`Must initializeQuoteMint before adding an oracle`);
 
+      const oracleKey = oracleInstance || Far('fresh key');
```

Comment:
I don't understand this "fresh key". Will be be fresh each time it's constructed?

Reply:
Yes.  Every call to `Far` creates a new object identity.  I was leaning on default arguments to the `Far` function, but have made it more explicit now.